### PR TITLE
feat: fast withdrawal from inactive pools

### PR DESCRIPTION
**Pull request by benr-ml**
## Description

Fast withdrawal from inactive pools

## Test plan

How did you test the new or updated feature?

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.
For each box you select, include information after the relevant heading that describes the impact of your changes that
a user might notice and any actions they must take to implement updates. (Add release notes after the colon for each item)

- [ ] Storage node:
- [ ] Aggregator:
- [ ] Publisher:
- [ ] CLI:

### CODE REVIEW

File: contracts/walrus/sources/staking/staking_inner.move [hunk 1]
```diff
@@ -409,7 +409,15 @@ public(package) fun request_withdraw_stake(
     _ctx: &mut TxContext,
 ) {
     let wctx = &self.new_walrus_context();
-    self.pools[staked_wal.node_id()].request_withdraw_stake(staked_wal, wctx);
+    let node_id = staked_wal.node_id();
+    self
+        .pools[staked_wal.node_id()]
```
Comment (karlwuest) [inline contracts/walrus/sources/staking/staking_inner.move]: Nit
```suggestion
        .pools[node_id]
```

File: contracts/walrus/sources/staking/staking_inner.move [hunk 2]
```diff
@@ -421,7 +429,16 @@ public(package) fun withdraw_stake(
     ctx: &mut TxContext,
 ): Coin<WAL> {
     let wctx = &self.new_walrus_context();
-    self.pools[staked_wal.node_id()].withdraw_stake(staked_wal, wctx).into_coin(ctx)
+    let node_id = staked_wal.node_id();
+    self
+        .pools[staked_wal.node_id()]
```
Comment (karlwuest) [inline contracts/walrus/sources/staking/staking_inner.move]: Nit:
```suggestion
        .pools[node_id]
```

File: contracts/walrus/sources/staking/staking_pool.move [hunk 3]
```diff
@@ -316,34 +318,59 @@ public(package) fun request_withdraw_stake(
...
+        // note that if the stake is in state Withdrawing, it can either be
+        // from a pre-active withdrawal, but then
+        // (in_current_committee || in_next_committee) is true since it was
+        // an early withdrawal, or from a standard two step withdrawal,
+        // which is handled below.
+    } else {
+        // normal two-step withdrawals
+        assert!(staked_wal.is_withdrawing(), ENotWithdrawing);
+        assert!(staked_wal.withdraw_epoch() <= wctx.epoch(), EWithdrawEpochNotReached);
+        assert!(activation_epoch <= wctx.epoch(), EActivationEpochNotReached);
```
Comment (karlwuest) [inline contracts/walrus/sources/staking/staking_pool.move:363]: Question: This last check is just a sanity check and should never be reached, right?

Comment (benr-ml) [reply contracts/walrus/sources/staking/staking_pool.move:363]: yes, it was there already and i think it's an invariant worth checking

**Pull request merged**